File: 02515. Shortest Distance to Target String in a Circular Array/solution.cs
/**
 * Problem: 2515. Shortest Distance to Target String in a Circular Array
 *
 * Difficulty: Easy
 *
 * Language: C#
 *
 * Performance: Runtime - 0 ms (Beats 100%)
 */

public class Solution
{
  public int ClosestTarget(string[] words, string target, int startIndex)
  {
    // Get length of words array
    int arrayLength = words.Length;

    // Initialize answer to maximum value
    int minSteps = int.MaxValue;

    // Iterate through all indices
    for (int i = 0; i < arrayLength; i++)
    {
      // Check if current word matches target
      if (words[i] == target)
      {
        // Calculate clockwise distance
        int clockwiseSteps = (i - startIndex + arrayLength) % arrayLength;
        // Calculate anticlockwise distance
        int anticlockwiseSteps = (startIndex - i + arrayLength) % arrayLength;

        // Update minimum steps
        minSteps = Math.Min(minSteps, Math.Min(clockwiseSteps, anticlockwiseSteps));
      }
    }

    // Return -1 if target not found, otherwise return minimum steps
    return minSteps == int.MaxValue ? -1 : minSteps;
  }
}

File: 02515. Shortest Distance to Target String in a Circular Array/solution.js
/**
 * Problem: 2515. Shortest Distance to Target String in a Circular Array
 *
 * Difficulty: Easy
 *
 * Language: JavaScript
 *
 * Performance: Runtime - 0 ms (Beats 100%)
 */

/**
 * Finds minimum steps to reach target word in circular array
 *
 * @param {string[]} words - Array of words
 * @param {string} target - Target word to find
 * @param {number} startIndex - Starting index
 *
 * @returns {number} Minimum steps or -1 if not found
 */
const closestTarget = (words, target, startIndex) => {
  // Get length of words array
  const arrayLength = words.length

  // Initialize answer to infinity
  let minSteps = Infinity

  // Iterate through all indices
  for (let i = 0; i < arrayLength; i++) {
    // Check if current word matches target
    if (words[i] === target) {
      // Calculate clockwise distance
      const clockwiseSteps = (i - startIndex + arrayLength) % arrayLength
      // Calculate anticlockwise distance
      const anticlockwiseSteps = (startIndex - i + arrayLength) % arrayLength

      // Update minimum steps
      minSteps = Math.min(
        minSteps,
        Math.min(clockwiseSteps, anticlockwiseSteps)
      )
    }
  }

  // Return -1 if target not found, otherwise return minimum steps
  return minSteps === Infinity ? -1 : minSteps
}
